internal/domain/interfaces.go

```diff
@@ -528,6 +532,81 @@ type UIEvent interface {
 	GetType() UIEventType
 }
 
+// A2ADirectService handles direct agent-to-agent communication
+type A2ADirectService interface {
+	// SubmitTask submits a task to a specific A2A agent in the background
+	SubmitTask(ctx context.Context, agentName string, task A2ATask) (string, error)
+
+	// GetTaskStatus retrieves the current status of a task
+	GetTaskStatus(ctx context.Context, taskID string) (*A2ATaskStatus, error)
+
+	// CollectResults collects completed task results
+	CollectResults(ctx context.Context, taskID string) (*A2ATaskResult, error)
+
+	// CancelTask cancels a running task
+	CancelTask(ctx context.Context, taskID string) error
+
+	// ListActiveAgents returns all currently active A2A agents
+	ListActiveAgents() (map[string]A2AAgentInfo, error)
+
+	// TestConnection tests connectivity to a specific agent
+	TestConnection(ctx context.Context, agentName string) error
+}
+
+// A2ATask represents a task to be submitted to an A2A agent
+type A2ATask struct {
+	ID          string                 `json:"id"`
+	Type        string                 `json:"type"`
+	Description string                 `json:"description"`
+	Parameters  map[string]interface{} `json:"parameters"`
+	Priority    int                    `json:"priority,omitempty"`
+	Timeout     int                    `json:"timeout,omitempty"`
+}
+
+// A2ATaskStatus represents the status of an A2A task
+type A2ATaskStatus struct {
+	TaskID       string           `json:"task_id"`
+	Status       string           `json:"status"` // pending, running, completed, failed, input_required
```

Comment:
```suggestion
	Status       string           `json:"status"` // pending, running, completed, failed, input_required
```

I think this could be an enum instead of the comment.